### PR TITLE
Feature/boto3

### PR DESCRIPTION
**Pull request by CyberDem0n**
Addresses https://github.com/zalando/stups-etcd-cluster/issues/3 and https://github.com/zalando/stups-etcd-cluster/issues/4

### CODE REVIEW

File: etcd.py [hunk 1]
```diff
@@ -426,36 +433,47 @@ def remove_unhealthy_members(self, autoscaling_members):
...
+        )
 
     def update_route53_records(self, autoscaling_members):
-        conn = boto.route53.connect_to_region('universal')
-        zone = conn.get_zone(self.hosted_zone)
+        conn = boto3.client('route53', region_name='universal')
+        zones = conn.list_hosted_zones_by_name(DNSName=self.hosted_zone)
+        zone = ([z for z in zones['HostedZones'] if z['Name'] == self.hosted_zone] or [None])[0]
         if not zone:
             return
```
Comment (a1exsh) [inline etcd.py]: Maybe `raise` something instead of returning silently?

File: etcd.py [hunk 2]
```diff
@@ -297,11 +297,16 @@ def get_my_instance(self):
...
-        res = conn.get_all_reservations(filters={'tag:{}'.format(EtcdMember.AG_TAG): me.autoscaling_group})
+        conn = boto3.resource('ec2', region_name=self.region)
+        instances = conn.instances.filter(
+            Filters=[{'Name': 'tag:{}'.format(EtcdMember.AG_TAG), 'Values': [me.autoscaling_group]}])
 
-        return [i for r in res for i in r.instances if i.state != 'terminated' and i.tags.get(EtcdMember.AG_TAG, '')
-                == me.autoscaling_group]
+        ret = []
+        for i in instances:
+            tags = {t['Key']: t['Value'] for t in i.tags}
```
Comment (a1exsh) [inline etcd.py]: This is used 3 times already, maybe worth making it a function?

**Comment by a1exsh** [comment]
I'm a bit alarmed that you change both code and the tests in the same commit(s) -- how do you know you're not breaking it with such a mix?  It would look safer if you change only the code, keeping the same tests, when change/improve the tests (if needed at all).

**Comment by CyberDem0n** [comment]
It's not possible to change code and test separately because API provided by boto3 is not the same as was in old boto and it requires different mocks...

File: etcd.py (hunk 2 above)
Comment (CyberDem0n) [inline etcd.py]: Yeah, make scene.

**Comment by a1exsh** [comment]
That's pity.  Anyway, looks good.

**Comment by feikesteenbergen** [comment]
+1

**Pull request merged**